junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertionUtils.java

```diff
@@ -93,7 +94,13 @@ private static String formatClassAndValue(Object value, String valueString) {
 	}
 
 	private static String toString(Object obj) {
-		return (obj instanceof Class ? getCanonicalName((Class<?>) obj) : String.valueOf(obj));
+		if (obj instanceof Class) {
+			return getCanonicalName((Class<?>) obj);
+		}
+		if (obj instanceof Object[]) {
+			return Arrays.toString((Object[]) obj);
+		}
+		return String.valueOf(obj);
```

Comment:
Wouldn't it be better to just use `org.junit.platform.commons.util.StringUtils.nullSafeToString(Object)` here?